code/parachain/frame/assets-registry/src/tests.rs

```diff
@@ -95,37 +107,43 @@ fn update_asset() {
 		let location = <Runtime as crate::Config>::ForeignAssetId::decode(
 			&mut &XcmAssetLocation::RELAY_NATIVE.encode()[..],
 		)
-		.unwrap();
-		let ratio = rational!(42 / 123);
-		let decimals = 3;
+		.expect("Location bytes translate to foreign ID bytes");
+		let asset_info = AssetInfo {
+			name: None,
+			symbol: None,
+			decimals: 4,
+			existential_deposit: 0,
```

Comment:
would this asset use native token as payment?

Reply:
Can you elaborate on what you're asking?

Reply:
If you set ED to 0. I have 1M of tokens. I can create 1M accounts? Never pay ED and spam network?

Reply:
yes

Reply:
we could have ED be a NonZeroU8 then. will this break `multi_existential_deposits`?

Reply:
on theory it should not. but afaik some benchamarks, 3rd party, did not worked well long time ago. i did not checked again.